sympy/sets/fancysets.py

```diff
@@ -9,9 +9,56 @@
 from sympy.core.sympify import _sympify, sympify, converter
 from sympy.logic.boolalg import And
 from sympy.sets.sets import Set, Interval, Union, FiniteSet, ProductSet
+from sympy.sets.conditionset import ConditionSet
+from sympy.utilities.iterables import flatten
 from sympy.utilities.misc import filldedent
 
 
+class Rationals(with_metaclass(Singleton, Set)):
+    """
+    Represents the rational numbers. This set is also available as
+    the Singleton, S.Rationals.
+
+    Examples
+    ========
+
+    >>> from sympy import S
+    >>> S.Naturals.intersect(S.Reals)
+    >>> S.Naturals0.intersect(S.Reals)
+    >>> S.Half in S.Rationals
+    True
+    >>> iterable = iter(S.Rationals)
+    >>> [next(iterable) for i in range(10)]
+    [0, 1/2, 2, -1/2, -2, 1/3, 3, -1/3, -3, 2/3]
+    """
+
+    is_iterable = True
+    _inf = S.NegativeInfinity
+    _sup = S.Infinity
+
+    def _contains(self, other):
+        if not isinstance(other, Expr):
+            return S.false
+        return other.is_rational
+
+    def __iter__(self):
+        from sympy.core.numbers import igcd, Rational
+        yield S.Zero
+        d = 2
+        while True:
+            for n in range(d):
+                if igcd(n, d) == 1:
```

Comment:
This misses the integers

Reply:
The integers are printed as `d/n` so `n,d`=`(2,4)` is skipped but (1,2) isn't and the (1,2) emits `1/2,2,-1/2,-2`. i.e., all integers are emitted when n is 1.